### ISSUE TITLE
[BUG] axolotl not launching

### ISSUE THREAD

**Issue by Spagett1** [nanu-c/axolotl#723]
### Description
axolotl will not launch when running with `go run . -e ut`
### Steps to Reproduce

1. install axolotl with the steps outlined here https://github.com/nanu-c/axolotl/discussions/615 
2. launch crayfish with `/usr/lib/crayfish &`
3. cd into the axolotl directory with `cd .local/src/axolotl`
4. set the library path with `LD_LIBRARY_PATH=/home/alarm/.local/src/axolotl`
5. run with `/usr/bin/go run . -e ut`

**Expected behavior:** [What you expect to happen]
axolotl should launch
**Actual behavior:** [What actually happens]
i recieve the error and the program crashes
#### Versions
latest commit in the main branch 
#### Device
pinephone running danctnix's arch 

#### Link to Debug Log
application outputs 
```
# github.com/mutecomm/go-sqlcipher
sqlite3.c: In function ‘sqlite3SelectNew’:
sqlite3.c:120891:10: warning: function may return address of local variable [-Wreturn-local-addr]
120891 |   return pNew;
       |          ^~~~
sqlite3.c:120852:10: note: declared here
120852 |   Select standin;
       |          ^~~~~~~
DEBU[0000] [axolotl] setup config                       
INFO[0000] [axolotl] Starting Signal for Ubuntu version 1.0.9 
INFO[0000] [axolotl] Setup completed                    
DEBU[0000] [axolotl] Run Backend                        
DEBU[0000] [axolotl] openDb: /home/alarm/.local/share/textsecure.nanuc/db 
INFO[0000] [axolotl] Axolotl-gui starting for sys: ut   
DEBU[0000] [axolotl] NewDataStore                       
INFO[0000] [axolotl] Axolotl server started             
INFO[0000] [axolotl] Starting axolotl ws                
DEBU[0000] [axolotl] Using axolotl-web path ./axolotl-web/dist 
DEBU[0000] [axolotl] NewDataStore finished              
INFO[0000] [axolotl] Loading Chats                      
INFO[0000] [axolotl] Db setup finished                  
DEBU[0000] [axolotl] Start Session after Decryption     
DEBU[0000] [axolotl] starting Signal connection         
DEBU[0000] [textsecure] loading config                  
DEBU[0000] [axolotl] config path:  /home/alarm/.config/textsecure.nanuc 
INFO[2022/01/16 00:07:32] [textsecure-crayfish] Starting crayfish-backend 
ERRO[2022/01/16 00:07:32] [textsecure-crayfish] crayfish not found textsecure doesn't work without crayfish 
ERRO[2022/01/16 00:07:32] [textsecure-crayfish] Please install crayfish, hints are in the README at https://github.com/signal-golang/textsecure 
exit status 1
``` 
when run
~/.cache/upstart/application-click-textsecure.nanuc_textsecure*.log does not exist

**Comment by nanu-c** [comment]
I am sorry for not looking into this earlier. Axolotl is currently not looking at `$PATH` for `crayfish` but at `./crayfish`
you need to place it next to the axolotl bin at `.local/src/axolotl`

**Comment by nanu-c** [comment]
@olof-nord maybe you more experienced how to do that.

**Comment by Laar3** [comment]
thanks, when i get a chance ill give that a shot

**Comment by olof-nord** [comment]
Reading through the guide you linked, I believe it needs some updating.

The prerequisites for Axolotl to run are three. axolotl-web js/css/html bundle, zkgroup and crayfish.

Looking at the log, the crayfish library is missing (or probably not in the expected place like nanu-c pointed out)

The reason why Axolotl is packaged as a snap, deb and Flatpak is that you as a user can install and get going easily.
If you choose to go the harder way and do the steps yourself, you need a deeper understanding of the project.

If you want to know what steps are required for building the application from source using Clickable, Snap, Flatpak, AppImage or deb, see the https://github.com/nanu-c/axolotl/blob/main/docs/INSTALL.md document.

If you want to build and install the parts manually, there are Makefile targets which should be configured to work out of the box.
`build-axolotl`, `build-axolotl-web`, `build-crayfish`, `build-zkgroup`.
For the install there are `install-axolotl`, `install-axolotl-web`, `install-crayfish` and `install-zkgroup` (I tried to name them logically)
Some build targets have prerequisites, such as `build-dependencies-axolotl` and `build-dependencies-axolotl-web`.

As for running the application, you could also simply run the axolotl binary directly with `./axolotl -e ut` instead of with go run.

**Comment by Symbian9** [comment]
I just tried to launch AppImage build of `1.2.0` under Debian 10, but it fails:

```
$ ./Axolotl-x86_64.AppImage
usr/bin/axolotl: error while loading shared libraries: libzkgroup_linux_x86_64.so: cannot open shared object file: No such file or directory
```

**Comment by olof-nord** [comment]
Thats probably true: the AppImage is missing the `crayfish` and `zkgroup` libraries, and also doesnt come with Electron bundled but tries to download it on initial startup.

The most up to date packaging options at the moment are Snap, Flatpak/Flathub and Deb.

**Comment by nuehm-arno** [comment]
Is this still an issue?

**Comment by Symbian9** [comment]
> Is this still an issue?

Yes:

```
$ ./Axolotl-x86_64.AppImage
usr/bin/axolotl: error while loading shared libraries: libzkgroup_linux_x86_64.so: cannot open shared object file: No such file or directory
```

**Comment by nanu-c** [comment]
this is solved in #985
